hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -952,4 +966,5 @@ def select(self):
             self.used_examples.append(result)
             return result
         else:
+            assert self.used_examples
```

Comment:
Any particular reason to have this assert here? The next line will error anyway if we would trigger it.